portal/static/js/main.js

```diff
@@ -2896,8 +2920,130 @@ var tnthDates = {
         }
         return toReturn
     },
-    "isDate": function(d) {
-        return Object.prototype.toString.call(d) === "[object Date]" && !isNaN( d.getTime() );
+    "isValidDefaultDateFormat": function(date, errorField) {
+        if (!hasValue(date)) return false;
+        if (date.length < 10) return false;
+        var dArray = $.trim(date).split(" ");
+        if (dArray.length < 3) return false;
+        var day = dArray[0], month = dArray[1], year = dArray[2];
+        //console.log("day: " + day + " month: " + month + " year: " + year)
+        if (day.length < 1) return false;
+        if (month.length < 3) return false;
+        if (year.length < 4) return false;
+        if (!/(0)?[1-9]|1\d|2\d|3[01]/.test(day)) return false;
+        if (!/jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec/i.test(month)) return false;
+        if (!/(19|20)\d{2}/.test(year)) return false;
+        var dt = new Date(date);
+        if (!this.isDateObj(dt)) return false;
+        else if (!this.isValidDate(year, this.convertMonthNumeric(month), day)) {
+            return false;
+        } else {
+          var today = new Date(), errorMsg = "";
+          if (dt.getFullYear() < 1900) errorMsg = "Year must be after 1900";
+          // Only allow if date is before today
+          if (dt.setHours(0,0,0,0) > today.setHours(0,0,0,0)) {
+              errorMsg = "The date must not be in the future.";
+          };
+          if (hasValue(errorMsg)) {
+            if (errorField) $(errorField).text(errorMsg);
+            return false;
+          } else {
+            if (errorField) $(errorField).text("");
+            return true;
+          }
+        };
+    },
+    "isDateObj": function(d) {
+        return Object.prototype.toString.call(d) === "[object Date]" && !isNaN( d.getTime());
+    },
+    "isValidDate": function(y, m, d) {
+        var date = new Date(y,parseInt(m)-1,d);
+        var convertedDate = ""+date.getFullYear() + (date.getMonth()+1) + date.getDate();
+        var givenDate = "" + y + m + d;
+        return ( givenDate == convertedDate);
+    },
+    /*
+     * NB
+     * For dateString in ISO-8601 format date as returned from server
+     * e.g. '2011-06-29T16:52:48'*/
+
+    "formatDateString": function(dateString, format) {
```

Comment:
It might be better to split this up into smaller functions that act on Date objects (instead of strings), separating the parsing and formatting steps and calling those smaller functions as necessary.

Passing around Date objects would only require you to parse the date string once, then they could be formatted in any number of ways. As far as I understand the Intl library acts on Date objects so using them as an intermediate format would be a step in the right direction.